spyder/widgets/sourcecode/base.py

```diff
@@ -277,6 +278,11 @@ def setup_completion(self):
     def set_indent_chars(self, indent_chars):
         self.indent_chars = indent_chars
 
+    def set_tab_stop_width_spaces(self, tab_stop_width_spaces):
+        self.tab_stop_width_spaces = tab_stop_width_spaces
+        self.setTabStopWidth(tab_stop_width_spaces
+                             * self.fontMetrics().width('9'))

```

Comment:
I think other character could work, `self.fontMetrics().width('9')` returns the width of a character in the editor

I use '9' because It is used in other places with the same purpose:
https://github.com/spyder-ide/spyder/blob/master/spyder/widgets/panels/edgeline.py#L41
https://github.com/spyder-ide/spyder/blob/master/spyder/widgets/panels/line_number.py#L171

Actually this magic it's pretty old https://github.com/spyder-ide/spyder/commit/f7c68f33574f2099f6cdab93762befda4d20bf7c#diff-a032958c9839f940f8038a099a3bf970R1072